### PR TITLE
ci: Extract `setup` job for `build` workflow

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Extract quick-running job that calculates matrix parameters for the `test` job

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]
we need to make that job mandatory, because if it fails in future, all depends will be skipped and allow merge

**Comment by zharinov** [comment]
Please let me test this PR more extensively...

**Comment by zharinov** [comment]
I've noticed that `ci:fulltest` should be added before PR creation, otherwise it won't work

**Comment by zharinov** [comment]
> we need to make that job mandatory, because if it fails in future, all depends will be skipped and allow merge

What if we make configuration `setup-test-matrix -> test [x15] -> finalize-test` and make `finalize-test` mandatory? Could it solve the problem?

**Comment by zharinov** [comment]
Fixed `ci:fulltest` 🎉

**Comment by viceice** [comment]
> > we need to make that job mandatory, because if it fails in future, all depends will be skipped and allow merge
> 
> What if we make configuration `setup-test-matrix -> test [x15] -> finalize-test` and make `finalize-test` mandatory? Could it solve the problem?

no, as finalize will be skipped, which is same as passed for GitHub automerge

**Comment by zharinov** [comment]
But we want to run only shards that were affected by PR, so I think it's worth to figure out how to make it work with automerge.

**Comment by zharinov** [comment]
Anyway, let's merge this one first, please 🙂

**Comment by zharinov** [comment]
BTW, I think I should better call it `schedule-tests` (or just `schedule`) rather than `setup-test-matrix`

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -29,7 +29,42 @@ env:
...
+        if: ${{ env.is-pr == 'true' }}
+        run: |
+          curl -s -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ env.repo }}/pulls/${{ env.pr }} \
+            | jq -c '{ labels: [.labels[].name] }' \
+            | sed '1s/^/pr=/g' \
+            > "$GITHUB_OUTPUT"
```
Comment (zharinov) [inline .github/workflows/build.yml:55]: image

🤦‍♂️

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
@rarkins please make the new job mandatory

**Comment by zharinov** [comment]
Making this mandatory is possible after merge and new CI run, if I remember correctly

**Comment by viceice** [comment]
> Making this mandatory is possible after merge and new CI run, if I remember correctly
can be done after it's run once. branch or PR doesn't matter. 😏

**Pull request merged**